bokehjs/src/coffee/models/transforms/customjs_transform.coffee

```diff
@@ -0,0 +1,31 @@
+import {Transform} from "./transform"
+import * as p from "core/properties"
+import {values} from "core/util/object"
+
+export class CustomJSTransform extends Transform
+  type: 'CustomJSTransform'
+
+  @define {
+    args:         [ p.Any,          {}       ] # TODO (bev) better type
+    code:         [ p.String      , ""       ]
+  }
+
+  @getters {
+    func: () -> @_make_func()
+    values: () -> @_make_values()
+  }
+
+  compute: (x) -> @func(@values..., require, x)
+
+  v_compute: (xs) ->
+    result = new Float64Array(xs.length)
```

Comment:
is this a safe assumption? What if users want to transform to colors or categorical values? In general I think we might want to require users to supply both a "scalar" and "vector" version of the transform, so that we don't have to make assumptions.